src/lucky_record/associations.cr

```diff
@@ -1,12 +1,19 @@
 module LuckyRecord::Associations
-  macro has_many(type_declaration)
+  macro has_many(type_declaration, foreign_key = nil)
     {% assoc_name = type_declaration.var %}
 
-    association table_name: :{{ assoc_name }}
+    association table_name: :{{ assoc_name }}, foreign_key: {{ foreign_key }}
 
     {% model = type_declaration.type %}
     def {{ assoc_name.id }}
-      {{ model }}::BaseQuery.new.{{ @type.name.underscore }}_id(id)
```

Comment:
This was returning `BaseQuery` rather than the model, but the tests weren't catching it. I changed it to return `results` which I think is the expected behavior

Reply:
Or actually is it supposed to return the `Query` so that it can be chainable. Like: `manager.employees.company_id(10)`?